sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -375,46 +549,30 @@ def __init__(
         ]
 
         self._timeout = timeout
-        self._client = None  # Will hold the Pipelineclient
-        self._operation = None  # Will hold an instance of LongRunningOperation
-        self._initial_response = None  # Will hold the initial response
-        self._pipeline_response = None  # Will hold latest received response
-        self._deserialization_callback = None  # Will hold the deserialization callback
         self._operation_config = operation_config
         self._lro_options = lro_options
         self._path_format_arguments = path_format_arguments
-        self._status = None
 
-    def status(self):
-        """Return the current status as a string.
-        :rtype: str
-        """
-        if not self._operation:
-            raise ValueError("set_initial_status was never called. Did you give this instance to a poller?")
-        return self._status
-
-    def finished(self):
-        """Is this polling finished?
-        :rtype: bool
-        """
-        return _finished(self.status())
-
-    def resource(self):
-        """Return the built resource."""
-        return self._parse_resource(self._pipeline_response)
-
-    @property
-    def _transport(self):
-        return self._client._pipeline._transport  # pylint: disable=protected-access
-
-    def initialize(self, client, initial_response, deserialization_callback):
+    def initialize(
+        self,
+        client: PipelineClientType,
+        initial_response: Any,
+        deserialization_callback: Callable[[Any], PollingReturnType_co],
+    ) -> None:
         """Set the initial status of this LRO.
 
-        :param initial_response: The initial response of the poller
+        :param client: The Azure Core Pipeline client used to make request.
+        :type client: ~azure.core.pipeline.PipelineClient
+        :param initial_response: The initial response for the call.
+        :type initial_response: ~azure.core.pipeline.PipelineResponse
+        :param deserialization_callback: A callback function to deserialize the final response.
+        :type deserialization_callback: callable
         :raises: HttpResponseError if initial status is incorrect LRO state
         """
         self._client = client
-        self._pipeline_response = self._initial_response = initial_response
+        self._pipeline_response = (  # pylint: disable=attribute-defined-outside-init
+            self._initial_response  # pylint: disable=attribute-defined-outside-init
```

Comment:
Is there a reason these weren't declared in the class attributes section? That seems to remove the need for the pylint suppression.

Reply:
Ah - I see that's because they are declared in the LROBasePolling in order to cleanly separate the async variants. This seems find so long as the `Any` typing for `initial_response` here doesn't negate any of the rest of this epic typing ;)

Reply:
Let me check everything is clean here, but I think once you actually instantiate the class, it does the right thing. There is a balance between the polling generic that do not know the types used (initial response could be whatever), and the implementation that needs the type.

TBH, if I were to write this code again, I wouldn't call it "initial response", it's biased toward HTTP. I would call it "initial context" or something like that.

Reply:
I re-typed them `PipelineResponseType`